internal/cloud/state.go

```diff
@@ -400,8 +400,17 @@ func (s *State) Unlock(id string) error {
 }
 
 // Delete the remote state.
-func (s *State) Delete() error {
-	err := s.tfeClient.Workspaces.Delete(context.Background(), s.organization, s.workspace.Name)
+func (s *State) Delete(force bool) error {
+
+	var err error
+
+	isSafeDeleteSupported := s.workspace.Permissions.CanForceDelete != nil
+	if force || !isSafeDeleteSupported {
+		err = s.tfeClient.Workspaces.Delete(context.Background(), s.organization, s.workspace.Name)
+	} else {
+		err = s.tfeClient.Workspaces.SafeDelete(context.Background(), s.organization, s.workspace.Name)
+	}
+
 	if err != nil && err != tfe.ErrResourceNotFound {
```

Comment:
You can see here that we ignore 404 errors when deleting the workspace, and I'm wondering if we should do the same thing when reading the workspace before deleting it in internal/cloud/backend.go...